pkgs/tools/system/osquery/default.nix

```diff
@@ -0,0 +1,82 @@
+{ pkgs }:
+
+with pkgs;
+let
+  buildStdenv = overrideCC stdenv llvmPackages.clangUseLLVM;
+  opensslArchive =
+    let
+      # https://github.com/osquery/osquery/blob/877d5e69ab97e15800b5c5128b3de2cf6f322942/libraries/cmake/formula/openssl/CMakeLists.txt#L3-L4.
+      version = "1.1.1q";
+      sha256 = "d7939ce614029cdff0b6c20f0e2e5703158a489a72b2507b8bd51bf8c8fd10ca";
+    in
+    fetchurl {
+      inherit sha256;
+      url = "https://www.openssl.org/source/openssl-${version}.tar.gz";
+    };
+in
+buildStdenv.mkDerivation rec {
+  pname = "osquery";
+  version = "5.5.1";
+
+  src = fetchFromGitHub {
+    owner = pname;
+    repo = pname;
+    rev = version;
+
+    fetchSubmodules = true;
```

Comment:
Most of those packages are present in nixpkgs. We don't want to compile against some vendored version especially with many security relevant packages. Please use our packages.

Reply:
In practice, i think it would be pretty hard to switch to our packages and i'm afraid this request could block this MR. Since nixpkgs fetches prebuild binaries, maybe we could also allow to fetch these submodules instead of using our own.

To provide some context, osquery uses their own dependencies to keep the support of old Linux distributions. So, if we switch to our own, we could also loose some osquery features.

Reply:
I already said we should not do it FWIW.